### PR TITLE
pull prettier into yarn scripts

### PR DESCRIPTION
**Pull request by outofambit**
## Overview

similar to #6204, but for prettier scripts

### CODE REVIEW

File: script/prettier.ts [hunk 1]
```diff
@@ -1,52 +0,0 @@
...
-
-  const fileList = result.stdout.toString().trim()
-
-  if (fileList.length > 0) {
-    console.log('These files are not formatted correctly:\n')
-
-    console.log(result.stdout.toString())
-
-    console.error(
-      chalk`{bold.green → To fix these errors, run {underline yarn lint:prettier --fix}}`
```
Comment (j-f1) [inline script/prettier.ts]: Similarly to https://github.com/desktop/desktop/pull/6204#discussion_r234801742, is this a valuable piece of information?

**Comment by shiftkey** [comment]
I've still got mixed feelings on this, because the prettier CLI isn't as friendly as `eslint`. Here's an example of a minor change that we currently catch:



That `→ To fix these errors, run yarn lint:prettier --fix` hint feels more intuitive than what's will happen after this PR is merged:



If I were to see this error in the build output of a PR, and I wasn't familiar with linting or prettier, it wouldn't be clear to me what to do to resolve it. We have [some docs](https://github.com/desktop/desktop/blob/master/docs/contributing/linting.md) on this now, which is cool, and we often have to walk contributors through the process still (we can now point them to docs about it :tada:), but I still miss having that friendliness present by default, which `eslint` does a bit better:

**Comment by shiftkey** [comment]
We talked a bit about this in Slack, and here's what this PR looks like after pulling in the next prettier release from source (`--check` now replaces `--list-different`):



Once we're on that version of prettier, I think we're in a better place to clean up ye old scripts.

**Comment by shiftkey** [comment]
<blockquote class="twitter-tweet" data-lang="en"><p lang="en" dir="ltr">We just released ✨Prettier 1.16✨ with HTML improvements and better CRLF handling!<a href="https://t.co/4kMm30QCDG">https://t.co/4kMm30QCDG</a></p>&mdash; Prettier (@PrettierCode) <a href="https://twitter.com/PrettierCode/status/1086897608805609472?ref_src=twsrc%5Etfw">January 20, 2019</a></blockquote>

**Comment by j-f1** [comment]
Yep, a couple bugs in the release. Ika’s planning to do a patch tonight and I should have time to open a PR here tomorrow.

**Comment by shiftkey** [comment]
@j-f1 are they bugs that affect `--check` or Desktop in general? I opened #6678 because that seemed to work locally, but happy to close that out if I should wait a bit

**Comment by j-f1** [comment]
[The milestone](https://github.com/prettier/prettier/milestone/26) says it’s just a couple issues with MDX, and a formatting issue with React Hooks-like function calls.

**Comment by shiftkey** [comment]
@j-f1 thanks for the context. I'm happy to leave #6678 as-is because it didn't actually change the formatting of our code, and it'll let us land this PR finally!

**Comment by outofambit** [comment]
once #6678 we can land this :D

**Comment by shiftkey** [comment]
Just merged `development` into this branch to clear the approval (also, conflicts). I think changing over the script to use `--check` is all that needs to be done here to close this out.

**Comment by romeo1563** [comment]
I need stop, how do this fucker

**Comment by shiftkey** [comment]
I went ahead and did the last bit of tidy up so we can close this out. This is ready for review!

**Comment by outofambit** [comment]
@iAmWillShepherd would you mind reviewing and doing the honors? thank you!

File: package.json [hunk 2]
```diff
@@ -27,9 +27,9 @@
     "clean:tslint": "rimraf tslint-rules/*.js",
     "compile:tslint": "tsc -P tslint-rules",
     "compile:script": "tsc -P script/tsconfig.json",
-    "lint": "yarn lint:src && yarn lint:prettier",
-    "lint:fix": "yarn lint:src:fix && yarn lint:prettier --fix",
-    "lint:prettier": "ts-node script/prettier.ts",
+    "lint": "yarn prettier && yarn lint:src",
+    "lint:fix": "yarn prettier --write && yarn lint:src:fix",
+    "prettier": "prettier --check \"./**/*.{yaml,yml}\" \"./app/**/*.{scss,ts,tsx,js,jsx,html}\" \"./script/**/*.{ts,js}\"",
```
Comment (kachkaev) [inline package.json]: I'm wondering if this pattern can be simplified down to something like `**/*.{ext1,ext2,extN}` and the exclusion of dirs completely moved to `.prittierignore`. This should make it easier to reason about what's prettified and also makes it more straightforward to configure [git hooks](https://prettier.io/docs/en/precommit.html). 🤔

Comment (outofambit) [reply package.json]: i like that idea a lot @kachkaev! i'll take a stab at that :)

File: package.json [hunk 3]
```diff
@@ -27,9 +27,9 @@
     "clean:tslint": "rimraf tslint-rules/*.js",
     "compile:tslint": "tsc -P tslint-rules",
     "compile:script": "tsc -P script/tsconfig.json",
-    "lint": "yarn lint:src && yarn lint:prettier",
-    "lint:fix": "yarn lint:src:fix && yarn lint:prettier --fix",
-    "lint:prettier": "ts-node script/prettier.ts",
+    "lint": "yarn prettier && yarn lint:src",
+    "lint:fix": "yarn prettier --write && yarn lint:src:fix",
+    "prettier": "prettier --check \"./**/*.{yaml,yml,scss,ts,tsx,js,jsx,html}\"",
```
Comment (kachkaev) [inline package.json]: If file extensions are put in alphabetical order, it's easier to extend the list later by adding new extensions. Just a though from a personal experience, things don't have to be like this.

**Comment by shiftkey** [comment]
```
$ prettier --check "./**/*.{yaml,yml,scss,ts,tsx,js,jsx,html}"
Checking formatting...
tslint-rules\buttonGroupOrderRule.js
tslint-rules\reactNoUnboundDispatcherPropsRule.js
tslint-rules\reactProperLifecycleMethodsRule.js
tslint-rules\reactReadonlyPropsAndStateRule.js
Code style issues found in the above file(s). Forgot to run Prettier?
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

These files are generated from `.ts` files in the same directory and don't exist in source control. These should be excluded from any linting checks.

**Comment by kachkaev** [comment]
Adding `tslint-rules/*.js` to `.prettierignore` will resolve this.

**Comment by outofambit** [comment]
all taken care of! thanks again @kachkaev!

**Pull request merged**